nix/shell.nix

```
@@ -38,6 +38,22 @@ in mkShell {
   LANG="en_US.UTF-8";
   LANGUAGE="en_US.UTF-8";
 
+  # to fix warnings like: ignoring the client-specified setting 'keep-derivations', because it is a restricted setting and you are not a trusted user
+  NIX_CONFIG = ''
```

Use [`readFile`](https://nix.dev/manual/nix/2.18/language/builtins#builtins-readFile) Nix builtin function:
```suggestion
  # Important to load our own Nix binary cache.
  NIX_CONFIG = builtins.readFile ./nix/nix.conf;
```

---

the core problem is the presence of these settings in `nix.conf`
These warnings only go away when I remove them from `nix.conf` and set them in a runtime way via `NIX_CONFIG` in `shell.nix`

---

That only happens because we use `NIX_USER_CONF_FILES` in the `Makefile`.

---

Ah that makes sense, Thanks I had missed out on that part.
Fixed in latest force push.